src/components/ModalWithForm.js

```diff
@@ -23,14 +34,14 @@ class ModalWithForm extends Modal {
   }
 
   setEventListeners() {
-    super.setEventListeners();
-
     this._modalForm.addEventListener("submit", (evt) => {
       evt.preventDefault(); // Prevent the default form submission
       const formData = this.getInputValues();
       this._handleFormSubmit(formData); // Correctly pass formData to the submission handler
       this._modalForm.reset();
```

Comment:
` this._modalForm.reset();`

You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.